src/cdomains/threadIdDomain.ml

```diff
@@ -30,21 +31,34 @@ sig
 
   module D: Lattice.S
 
-  val threadenter: t * D.t -> Node.t -> varinfo -> t list
-  val threadspawn: D.t -> Node.t -> varinfo -> D.t
+  val threadenter: t * D.t -> Node.t -> int option -> varinfo -> t list
+  val threadspawn: D.t -> Node.t -> int option -> varinfo -> D.t
 
   (** If it is possible to get a list of unique thread create thus far, get it *)
   val created: t -> D.t -> (t list) option
 end
 
 
+
 (** Type to represent an abstract thread ID. *)
 module FunNode: Stateless =
 struct
-  include Printable.Prod (CilType.Varinfo) (Printable.Option (Node) (struct let name = "no node" end))
+  include
+    Printable.Prod
+      (CilType.Varinfo) (
+      Printable.Option (
+        Printable.Prod
+          (Node) (
+          Printable.Option
+            (Printable.Int)
+            (struct let name = "no index" end)))
```

Comment:
MallocWrapper used `Chain` for this, why doesn't that work here?

Reply:
The problem is that `Chain` is a functor, returning a different module for each `n`. For the query we need a single module, hence this `Int` lattice.

Now that you mention it though, it might be possible to put the thread wrapper chain in a seperate module and reference it in both the thread wrapper analysis and in queries, which would clean things up.